StackExchange.Profiling.Wcf/WcfMiniProfilerClientInspector.cs

```diff
@@ -121,6 +123,8 @@ public void AfterReceiveReply(ref Message reply, object correlationState)
 
                     profiler.AddProfilerResults(resultsHeader.ProfilerResults);
                 }
+
+                profiler.Head?.Stop();
```

Comment:
Why are you relying on `Head`, you can pass the Step as part of the `CorrelationState` and use it. I think it's a safer way.

Reply:
Fixed